### PR TITLE
Filter wrapper data to use correct compilers in OpenMPI

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by tgamblin** [comment]
@davidbeckingsale: Looks good!  Thanks!  Can you fix the couple of minor comments I put on here?

**Comment by davidbeckingsale** [comment]
Yup, F77 is deprecated. I'll fix that up.

**Comment by tgamblin** [comment]
If it's deprecated but still there and used by projects, might be good to keep it in there and have the variables set (and the filters done) just in case.  Does the current OpenMPI not build with f77 enabled by default? If not, or if it's there in prior versions, might be good to check for the f77 `data.txt` files and filter them if they're there, and to just set the `OMPI_F77` env var regardless of whether it's used...

**Comment by davidbeckingsale** [comment]
Yeah, it's deprecated from 1.7 https://www.open-mpi.org/faq/?category=mpi-apps#override-wrappers-after-v1.0 but we can set it anyway.

I'm going to go through staging and building an older version too, just to make sure.

**Comment by tgamblin** [comment]
Thanks! looking at that document it looks like we should at least filter and support both `mpif77` and `mpif90` if they're there.  Users' builds likely still use these names.

**Comment by davidbeckingsale** [comment]
Yeah, the names are still there and just symlinked in the newer versions.
What the best way to filter different files depending on version?

**Comment by tgamblin** [comment]
Just check if it exists.  If it's not there, skip it

**Comment by davidbeckingsale** [comment]
They always exist, but they are symlinks in the newer versions.

**Comment by tgamblin** [comment]
I see you just checked whether it was a link instead -- that looks reasonable to me!  Want me to merge this or are you still working on it?

**Comment by davidbeckingsale** [comment]
I'm just checking that my updates work for both 1.6.5 and 1.8.2, will let you know in ~30 mins

**Comment by tgamblin** [comment]
@davidbeckingsale: awesome.  Thanks!

File: var/spack/packages/openmpi/package.py [hunk 1]
```diff
@@ -39,3 +51,55 @@ def install(self, spec, prefix):
...
+
+    def filter_compilers(self):
+        """Run after install to make the MPI compilers use the
+           compilers that Spack built the package with.
+
+           If this isn't done, they'll have CC, CXX and FC set
+           to Spack's generic cc, c++ and f90.  We want them to
+           be bound to whatever compiler they were built with.
+        """
+        kwargs = { 'ignore_absent' : True, 'backup' : False, 'string' : False }
```
Comment (davidbeckingsale) [inline var/spack/packages/openmpi/package.py]: I'm still ending up with ~ files even though backup is set to False.

**Comment by davidbeckingsale** [comment]
This works for me, so feel free to merge.

**Comment by tgamblin** [comment]
Thanks!

**Pull request merged**